observatory-platform/observatory/platform/utils/template_utils.py

```diff
@@ -122,13 +122,37 @@ def upload_files_from_list(files_list: List[str], bucket_name: str) -> bool:
         blob_names.append(blob_name(file_path))
 
     success = upload_files_to_cloud_storage(bucket_name, blob_names, files_list)
+    if not success:
+        raise AirflowException()
+
     return success
 
 
+def add_partition_date(list_of_dicts: List[dict], partition_date: datetime,
+                       partition_type: bigquery.TimePartitioningType = bigquery.TimePartitioningType.DAY):
+    """ Add a partition date key/value pair to each dictionary in the list of dicts.
+    Used to load data into a BigQuery partition.
+
+    :param list_of_dicts: List of dictionaries with original data
+    :param partition_date: The partition date
+    :param partition_type: The partition type
+    :return: Updated list of dicts with partition dates
+    """
+    if partition_type == bigquery.TimePartitioningType.HOUR:
+        partition_date = partition_date.isoformat()
+    else:
+        partition_date = partition_date.strftime('%Y-%m-%d')
+
+    for entry in list_of_dicts:
+        entry[partition_field] = partition_date
```

Comment:
This should come from a parameter in the function, as you may want to change what field is the partition for a specific table.